etc/builtin.conf

```diff
@@ -97,3 +97,16 @@ sub-shadow-offset=4
 [box]
 profile=osd-box
 profile=sub-box
+
+[image]
+profile-cond=(get('current-tracks/video', {}).image and not get('current-tracks/video', {}).albumart) or (not get('current-tracks/video') and get('user-data/mpv/image'))
```

Comment:
You can override options in the profile by defining an image profile in mpv.conf, as is documented.

Already many fundamental things don't work without lua support, osc, console, ytdl_hook. You can still play videos basically without Lua regardless. Without lua you will have to manually apply the profile with a key binding.

Only key bindings already useless for images are different.

The whole point of including a builtin conditional profile is to make mpv a good image viewer out of the box, else we could just put a link to my config. Users can define extra image options without having to deal with this convoluted condition, and without having to recommend them to use enable-section and disable-section, only to remove later them.

Reply:
> < and > still work for images, these just extend default key bindings. No key binding that is useful for both videos and images (e.g. stats, gamma) is changed.

Thanks, that answers my concerns then. I'm fine with the current approach.

Reply:
And prevents users from modifying those key binds. It is more user-friendly to provide input.conf line config for users to use/modify.

Reply:
> It still works if we use `enable-section image` in the script. It's already in the current autoprofile implementation anyway. `image-input.conf` is always loaded and the user can modify the config file.

Well you said

> This would also avoids the usage of input section commands entirely because you can just use mp.add_key_binding instead.

using script to manage sections instead of autoprofile is invariant to me.

Reply:
> This is even more problematic because it treats a profile name specially that may cause unexpected behavior. For example if I already have a profile called "image" that I only use when I want (by aliasing image viewer command to mpv --profile=image for example) it will now be loaded no matter what.

Please submit a PR removing all the profiles defined here https://github.com/mpv-player/mpv/blob/master/etc/builtin.conf

Reply:
> It's easy to do if they're implemented in a script.

It's already easy to add an option and check it in the profile condition, it just not needed, because most users don't need to disable every single option. Of course, it can be added if there is a consensus that it's needed.

> It's no less controversial because it affects loading of non-images. If no one has a problem with it just make it default.

I don't have commit rights to change any default, and It doesn't affect loading non-images in any meaningful away, of course it's less controversial. It's up to maintainers if they want to change the global default, which I'm fine with. It is not very useful for videos anyway, so I don't think it matters.

> It's common for the next entry to be a video if you use autocreate playlist by default and the folder has both videos and images.

Um, how does --autocreate-playlist add youtube videos to the playlist? But as I said, it doesn't cause disadvantages anyway.

> Can't you just use page up/down instead? They are "useless" keys for images. sxiv isn't some kind of gold standard that we must follow.

I don't think pressing p to pause instead of space is such a vital feature that we need to have different key bindings specifically for that, I find it useless in the default config too, because you can press space more easily. I like n and p both because they are near hjkl so you can use most key bindings with one hand, and because you can bind them with shift to navigate sub-playlists so they are variations of the same key bindings. Page up and down are too far away from other keys for what is the most common action in image viewers. In fact, feh and imv bind left and right arrows to navigate the playlist, and feh even binds n and p on top of arrows. We can always bind both, though. More feedback is welcome, if someone has an opinion on what to bind playlist-next and prev to.

> Either way it's better than the autoprofile implementation. We can weigh the trade off between input sections and key binding configs.

Enabling different options based on conditions is the exact use case of autoprofiles, and lets users modify the profile in mpv.conf like any other in the standard way instead of using script-opts, using a different script is reinventing the wheel. The only reasons you provided to use a different script is that it won't use enable-section, but then said that it should use it anyway, and that it can be disabled easily, when it's equally easy to add an option and check `profile-cond=some_option and get('current-tracks/video/image')...` in the profile.

Reply:
> Sorry to say but this is just total BS excuse. You can just open a PR for this to change the default and ask for feedbacks. Maintainers can merge it if the change is wanted. I have already mentioned how this affects the loading of non-images.

I said I already proposed changing the default in IRC and asked for feedback. We usually make initial proposals for changes in IRC, not Github. At least sfan5 didn't want to change it. If he changes his mind and no one opposes it, we can enable it by default.

> If you want it to really only be applied to images, again a script is better, as it can just check the filename of the next item in the playlist and make a guess whether to enable it or not.

I don't REALLY want it applied to images only, it is only you who are obssessing over what exactly it applies to when it doesn't make any meaningful difference. Guessing by filename which can be wrong when prefetching doesn't cause any harm is ridiculous.

> It will do for "local" videos, which can be a mounted network file share. On your own admission "It is not very useful for videos anyway" so it shouldn't be enabled. If it "doesn't cause any disadvantages", as I said open a PR to change the default.

Mounted network file shares are not ytdl_hook videos, and not very useful doesn't it mean that it's harmful.

> I like page up and page down both because they are near arrow keys so you can use most key bindings with one hand, and because you can bind them with shift to navigate sub-playlists so they are variations of the same key bindings.
>
> mpv in general doesn't use "vim-like" navigation pattern, and using page up/down with arrow keys has all the advantages you mentioned while being more similar to video keys and not taking up `p`.

n and p are not vim keybindings, and mpv does have shift key bindings that are variations of the base key bindings. Page up and down are not as close to arrow keys as other keys are to one another, which is why I doubt any other image viewer uses those keys for navigation primarily. We can also bind both, if preferred. As I said I don't think not taking up p matters, because you normally pause with space, and because pausing is only marginally useful in slideshow mode, and you can just disable the slideshow instead, that's how you do it in image viewers, they don't have pausing (of course, I plan of proposing to default to --image-display-duration=inf once image viewing works well by default).

> A script can still apply profiles with no problem. It also allows handing mouse better because it can separate clicking and dragging unlike `input.conf`.

This profile doesn't have to separate any clicking from dragging, it is only a list of options.

> Then add an option for that, which would solve my biggest problem with this.

Great, but you're not the one making decisions. I find it enough to override individual options and another option on top of that bloat, if maintainers deem it useful, then I will add it.

Reply:
Actually, you can just override the profile what an empty `profile-cond` to disable it. Added it to the docs. We don't need to bloat the global options.

Reply:
FWIW I share the general concerns mentioned here and think we should leave an image adjusted profile to users

> Have you asked/informed @sfan5 about the current situation where it's kind of enabled for images only but not really? What was his reason for not changing it by default? I would say if his reason isn't solely for ytdl_hook, it would apply to the current situation too, because it doesn't affect image only.

```
[2024-11-06 17:51:21] <guido> should --prefetch-playlist also default to yes? some users are requesting it
[2024-11-06 17:51:53] <sfan5> no
[2024-11-06 17:52:22] <guido> what disadvantage does it have?
[2024-11-06 17:53:48] <sfan5> "Highly experimental" ???
[2024-11-06 17:54:10] <guido> that was written almost 8 years ago
[2024-11-06 17:54:15] <guido> it's not experimental anymore
[2024-11-06 17:54:37] <sfan5> by what measure? has the code changed?
[2024-11-06 17:54:43] <sfan5> "This  can  give  subtly  wrong  results  if  per-file options are used" does not inspire confidence either
[2024-11-06 17:54:56] <guido> by the measure that people use it and no issue was ever reported
[2024-11-06 17:56:45] <sfan5> ok so nobody appears to hit the documented edge case
[2024-11-06 17:56:58] <sfan5> still I would not enable something by default that we know can cause problems
[2024-11-06 18:00:02] <guido> I'm skeptical of whether it can really cause problems or wm4 was just fearmongering
```

Reply:
> You said "there was not much feedback" so I assumed there wasn't opposition. Have you asked/informed @sfan5 about the current situation where it's kind of enabled for images only but not really? What was his reason for not changing it by default? I would say if his reason isn't solely for ytdl_hook, it would apply to the current situation too, because it doesn't affect image only.

I did not think it would be important to elaborate at the time. He didn't provide a specific reason other than that it may cause problems because of the wording in docs. Enabling only for images first can even be a good way to prove that there are in fact no problems.

> It is actively harmful. Why should mpv make my file server spin and waste network traffic loading 1GB of cache for the next file when I view an image?

You misunderstood how the option works, it can't load 1GB of cache because it only opens a connection to the next file. Try it with a video, and you'll see it starts with an empty cache. It is more useful for images since there is nothing to prefetch after the first frame. The reason for prefetching is obviously to open the next file faster.

> ?????? They're literally next to the up arrow on all laptops I have, and even for full-sized keyboard it's not far away. Also JPEGView does use page up/down for navigation.

Ok, I was thinking of desktops since I typically use those. But as I wrote above I remembered later that we also bind Enter to playlist-next, so you can already also use that for a binding near arrows, at for playlist-next, it is even closer on my laptop.

> Can I toggle with only mouse? MBTN_RIGHT is occupied and no longer toggles.

I don't have a solution for that other than not binding drag-to-pan, ideally we can move cycle pause to MBTN_LEFT, but it is problematic to implement the deadzone on macOS:

2024-11-23 13:20:10     der_richter     mmh it seems like that mechanism of vo dragging is incompatible with the way macOS' dragging works. eg the flag has to be set before a mouse down event starts, though VOCTRL_BEGIN_DRAGGING will only be send after the event. activating the dragging then will only work with the next left click down and not the current event.

> This isn't answering the question.

There was no question? I don't understand what separating clicking from dragging has to do with a list of options.

> This is nonsense excuse and backwards development process if you have never asked them about this and assumed that they won't accept this and use this as a source of authority for the sake of argument.

When did I assume they won't accept anything if I literally wrote that if maintainers deem it useful I will add it? I'm just not going to change things I don't find useful only because you told me to before receiving more feedback. Anyway, as I wrote later I realized you can just override `profile-cond=` to disable the profile.

Reply:
Apparently --prefetch-playlist used to also cache the stream: https://github.com/mpv-player/mpv/issues/6753. Probably it could have caused more issues back then which is when its docs were written, but not now that it doesn't do much.

In the docs per-file options refers specifically to file-local-options, which conditional profiles aren't exactly, this profile is even written so that it doesn't reset while changing file. But still  file-local-options + prefetch-playlist work fine for me, and nobody reported issues with them.

The only issue I know as I wrote before is that it uses the the demuxer probing options set before prefetching if changed at after prefetching,  e.g. if you `set demuxer mkv`, I don't think anybody changes these at runtime and that it is worth giving up faster image loading by default for.

Reply:
> FWIW I share the general concerns mentioned here and think we should leave an image adjusted profile to users

Can you specify which ones exactly? Most of the concerns aren't really valid and the discussion keeps jumping from one point to another after guido addresses one.

The original concern was that the user can't disable the profile autoloading, but now it's established that it can be disabled with `profile-cond=`. The bindings point is also not really valid, since all the video-mode bindings that are being changed are useless on images anyway.

Reply:
> The effect is the same when the condition changes between files, regardless whether the options are reset between files or not. In both cases, some options are applied to one file but not to another.

There is no effect if the condition stays true between files, and every instance of per-file in the docs refers specifically to file-local-options.

> Then there should be no problem enabling it by default. But @sfan5 didn't want to enable it unconditionally, while also wasn't clear about whether the "image adjusted profile" should be enabled by default or not. Either way to me he isn't showing a great confidence about enabling this option. I think this should wait until there is clear answer about it.

I don't care about enabling it by default, if you want to enable it by default, you open a PR for it.

The option is from 2017, I don't know you expect a clear answer to suddenly magically appear. #15378 eases the scary warnings.

Reply:
It is clear that wm4 said per-file options he meant file-local-options, even the heading of the section documenting them is called Per-File Options. Whether the condition stays true or false is unrelated to file-local-options, unless you append to reset-on-next-file in the condition, which doesn't work correctly anyway because it's set too late. And I used prefetch-playlist with both file-local-options and conditional profiles every day since wm4 added conditional profiles 4 years ago and never saw a single issue. So even if it broke regular options, I would have noticed by now, or someone else would have reported it.

Great for your stance, but I only care about it for images, and it is easier to get a smaller change merged by applying it to images, where it is much more useful.

@sfan5 These are just more useful default for images, the same way we provide defaults for videos, without better defaults most users will never realize that mpv is a great image viewer, think about my mpv-console-select repo which had only 3 stars, and only after I linked it in an issue, but only after it was upstreamed many people are saying that select.lua is very useful, and you can disable the profile, so why make them manually enable it? Do you oppose something specifically?

Reply:
> But @sfan5 didn't want to enable it unconditionally, while also wasn't clear about whether the "image adjusted profile" should be enabled by default or not. Either way to me he isn't showing a great confidence about enabling this option. I think this should wait until there is clear answer about it

My concerns about `prefetch-playlist` source only from the documentation implying that it is broken in some way or another.
If isn't the experimental feature it is made to sound like then I have no concerns.

> @sfan5 These are just more useful default for images, the same way we provide defaults for videos, without better defaults most users will never realize that mpv is a great image viewer, [...] Do you oppose something specifically?

it feels like scope creep to me. are we going to add dedicated audio file bindings too? (no)
but here are some concrete problems:
* mpv is not actually a great image viewer because it falls apart if the image is bigger than the GL max texture size
* ^ and it doesn't work with [tiled HEIC pictures](https://trac.ffmpeg.org/ticket/11170) that Samsung or iOS phones produce by default
* "I don't think this is good behavior, especially it's very confusing when image keybinds are completely different from normal ones"
* I don't have better suggestion but the 'input.conf' vs 'input-image.conf' feels kinda clunky. it's no longer a single file as the definite source of keybindings mpv reads.
* "Also it doesn't work if mpv isn't compiled with lua support." this is minor but I think it would be cleaner to do the `image` (un)applying decision in C code

Reply:
> mpv is not actually a great image viewer because it falls apart if the image is bigger than the GL max texture size

libplacebo supports sliced texel uploads on d3d11 and vulkan, and mpv no longer defaults to opengl on any platform so this shouldn't be an issue. https://code.videolan.org/videolan/libplacebo/-/merge_requests/542

> doesn't work with [tiled HEIC pictures](https://trac.ffmpeg.org/ticket/11170) that Samsung or iOS phones produce by default

neither does any image viewer using imlib2 (feh/imv/sxiv) without patching imlib2 with external patches as far as I can tell, and mpv supports way more formats than other traditional image viewers.

> "I don't think this is good behavior, especially it's very confusing when image keybinds are completely different from normal ones"

They are _not_ completely different. Only keys mentioned in input-image.conf are changed, the rest remain unchanged. And the only keys changed are keys that are mostly completely used on image files anyway.

> I don't have better suggestion but the 'input.conf' vs 'input-image.conf' feels kinda clunky. it's no longer a single file as the definite source of keybindings mpv reads.

They can be moved to the same file using input sections, I don't have an opinion on this.

> "Also it doesn't work if mpv isn't compiled with lua support." this is minor but I think it would be cleaner to do the image (un)applying decision in C code

I was also of this opinion with preferring the C implementation of cursor-centric-zoom, but I was told it doesn't matter because nobody builds mpv without Lua.

I had a whole section to this response about the pros of using mpv as an image viewer, but I wasn't sure if I needed to or not. But I'll just say that mpv is the best image viewer on Wayland that isn't using xwayland. Let me know if you're unsure why me and @guidocella and many others have been using mpv as an image viewer for a while already, and why we'd like for mpv to support it out of the box.

Reply:
> WIth the current PR, does `prefetch-playlist` change from yes to no when I switch from image to non-image files, after the condition is evaluated, which causes the next file to not use prefetch, even when it's an image? WIth the current PR, does `prefetch-playlist` change from no to yes when I switch from non-image to image files, after the condition is evaluated, which causes the next file to use prefetch, even when it's a non-image?

Yes, so what's your point, that there are flaws? I already said the profile is trying to provide useful defaults and not to be perfect and that there is no harm in prefetching videos.

> Stop beating around the bush with factually true but totally irrelevent claims. Conditional profile is not mentioned in the documentation of `prefetch-playlist` because `prefetch-playlist` predates conditional profiles, and when conditional profile was implemented he forgot to update `prefetch-playlist` documentation.

He may have, he may not, there is no proof either way. But file-local-options are definitely implemented differently.

> There should be no issue enabling it globally then. After all, "someone else would have reported it" if it doesn't work. If it has no issue enabling it by default of course you won't notice anything "broken" when you use in conditional profiles. But not being "broken" doesn't mean it's working as intended to be applied to only image files.

And I already said I'm for enabling it by default, and will try to fix the demuxer option issue. If there is a consensus that it should be the default, great, if there is no consensus, then I'm not going argue for it again after having already done it in IRC, because I think it's mostly useful for images, and don't think prefecthing the next video after images is as invasive as always enabling it. So if there is no consensus and you want to make it the default, it is up to you to argue for it by opening a PR. If you don't want to make it the default, then you are no one to tell me to do, so stop nagging by repeating the same points.

> it feels like scope creep to me. are we going to add dedicated audio file bindings too? (no) but here are some concrete problems:

Video key bindings already work equally well for audio files, but not for images. We do in fact have features specifically for audio files in cover arts and playing in the terminal only, so that is a precedent for having features specifically for images too.

> * mpv is not actually a great image viewer because it falls apart if the image is bigger than the GL max texture size

I never saw even one of these images bigger than GL max texture sizes in years of real world usage, so I don't think that alone makes it bad. In general large images have much better performance in gpu-next since https://code.videolan.org/videolan/libplacebo/-/merge_requests/542. According to llyyr they even work actually, but even they didn't it would be fine to use a different image viewer if you need them, it doesn't have to be preferred by everyone. I bet these don't work either in simpler programs like sxiv and feh, which are beloved regardless.

In general mpv is my favorite image viewer because (mostly taken from my image config repo's README):

- It's feature rich: over 1000 options, can play image URLs, play from pipes, etc.
- It's extremely extensible with scripts, much more than any image viewer, see occivink's scripts, Dudemanguy's manga script, or the image mode in Samillion's modernz, which image viewer lets you change it so much so easily?
- Advanced scaling, including shaders
- HDR/color managment (I don't use this, but it's commonly cited as a strong point)
- quit-watch-later with image galleries and manga
- Can browse directories with both videos and images in the same program, seamlessly adapting key bindings
- Background transparency
- Multiplatform, in particular Wayland support, unlike sxiv or feh. I like imv less than sxiv, and it is not maintained, so mpv is the definitely the best on Wayland
- It can even navigate between images faster than simpler programs like sxiv with --prefetch-playlist
- You're already familiar with it from playing video and audio

https://github.com/occivink/mpv-image-viewer is fairly popular, and we already had requests for better support for image viewing out of the box in https://github.com/mpv-player/mpv/issues/7983 and by some IRC users, like BillTorvalds who wanted to use it but found it too hard to setup.

> * ^ and it doesn't work with [tiled HEIC pictures](https://trac.ffmpeg.org/ticket/11170) that Samsung or iOS phones produce by default

See llyyr's response.

> * "I don't think this is good behavior, especially it's very confusing when image keybinds are completely different from normal ones"

See llyyr's response, image key bindings extend the regular ones by replacing keys useless for images, like seeking and volume, with useful ones like panning and zooming, keeping key bindings useful for both, with the minor exception of p. I actually did try to bind i to zoom in images like in imv years ago, and realized it was harder to use, so e.g. stats is still on i.

> * I don't have better suggestion but the 'input.conf' vs 'input-image.conf' feels kinda clunky. it's no longer a single file as the definite source of keybindings mpv reads.

Agreed. image-input.conf is a workaround to not use deprecated input sections, which let you specify key bindings with {image} in the same input.conf. I tried to undeprecated them in https://github.com/mpv-player/mpv/pull/14050, but it was rejected. So if there is a consensus to undeprecated them, I'm all for it.

> * "Also it doesn't work if mpv isn't compiled with lua support." this is minor but I think it would be cleaner to do the `image` (un)applying decision in C code

When I implemented cursor-centric-zoom in C people said that using Lua is better, then when I use Lua people say that features must work without Lua?? But surely, just using default keybindings or configuring them yourself for images without Lua can't be worse than no osc/console/ytdl_hook.

Reply:
Just to provide some feedback from a user's pov: I'm generally fine with the behavioral changes, as long as there is a clean and easy way to disable them.

I agree with @na-na-hi about key bindings changing across files being inconsistent, and I for one don't like this behavior. That said, the benefits of having more useful bindings for images are there, and there are certainly users who will like this change.

My only real concern is using `profile-cond` in a built-in profile:
* Having to put `profile-cond=` in the config file to disable the profile looks too much like a hack. We can have a dedicated option to control whether the `image` profile should be applied, like how `player-operation-mode` controls `builtin-pseudo-gui`. (Or, on second thought, introduce something like `{force,ignore}-builtin-profiles=profile1,profile2` in case more conditional built-in profiles are added in the future.)
* As someone who happens to have playlists with videos, audios and images mixed together, `image` profile options temporarily affecting videos and audios is a real problem for me. For this, implementing the decision in C is clearly better.

Reply:
> I agree with @na-na-hi about key bindings changing across files being inconsistent

The only keybindings that are changing are ones that are completely useless on an image anyway! What usefulness does volume up/down have on an image file with no audio tracks? or what usefulness does `u` have on an image file that won't have any subtitle tracks? Or what usefulness does seeking with arrow keys have on an image file with 1 frame? Or what usefulness does r/R moving subtitles have on an image file that, once again, have no subtitles?

For the umpteenth time, `image-input.conf` is only changing keybinds listed in that file. The rest are the same as regular defaults. I'd request @na-na-hi to stop pretending to misunderstand this point or leave this discussion, as it has caused confusion for @sfan5 and @sunpenghao both. It's much harder to fix damage done by misinformation than it is to spread misinformation.

> Having to put profile-cond= in the config file to disable the profile looks too much like a hack. We can have a dedicated option to control whether the image profile should be applied, like how player-operation-mode controls builtin-pseudo-gui. (Or, on second thought, introduce something like {force,ignore}-builtin-profiles=profile1,profile2 in case more conditional built-in profiles are added in the future.)

This is trivial to do and not a blocker for this. The only reason there isn't an option is because once again, someone mentioned that an option for this would be bloat.

> As someone who happens to have playlists with videos, audios and images mixed together, image profile options temporarily affecting videos and audios is a real problem for me. For this, implementing the decision in C is clearly better.

To be honest, I have no idea what you're getting this from. The discussion at hand is `prefetch-playlist` causing `--demuxer-*` options to affect the next file even if they shouldn't be applied because the `profile-cond` isn't satisfied anymore. There's ways to improve this situation, by invalidating prefetch if demuxer options are changed. But it's also incredibly uncommon to change `--demuxer` options during runtime specifically when you're viewing an image and the next file is a video.

Reply:
An option just does the same thing in a more roundabout away, you have to read the docs to see how to disable it either way, and the expectation is that the vast majority of users don't need to disable everything in the profile at once as it generally provides better defaults for images.

auto_profiles.lua already evaluates profiles within synchronous hooks, so applying the profile from C wouldn't magically make it more synchronous as far as I know, it just does the same thing with more code and more risk of memory issues.

This config is specifically meant to be used with playlists with videos and images mixed together, which I use too. None of the options cause problems if the next file is not an image. (autoload-fies=no would have caused them, it worked before because there was a mistake in the non-image condition which made options restore on each file, so I removed it).

Reply:
> The only keybindings that are changing are ones that are completely useless on an image anyway! What usefulness does volume up/down have on an image file with no audio tracks? or what usefulness does `u` have on an image file that won't have any subtitle tracks? Or what usefulness does seeking with arrow keys have on an image file with 1 frame? Or what usefulness does r/R moving subtitles have on an image file that, once again, have no subtitles?

What about holding `RIGHT` to skim over a video and when EOF has been reached and the player jumps to the next playlist item which is an image, the key suddenly changes its behavior? Don't get me wrong though, I have no intention to make any case on this. Like I said, there are people who will like the new bindings.

> The only reason there isn't an option is because once again, someone mentioned that an option for this would be bloat.

That's why I proposed the `{force,ignore}-builtin-profiles` thing. What if after this developers decide to add another profile for audios?

> To be honest, I have no idea what you're getting this from. The discussion at hand is `prefetch-playlist` causing `--demuxer-*` options to affect the next file even if they shouldn't be applied because the `profile-cond` isn't satisfied anymore.

And I'm not talking about `prefetch-playlist`. My point is implementing conditional options intended for one type of files in a way that may have side effects for other types is flawed. It's true that options like `script-opt=osc-deadzone=1` or `video-aspect-override=no` don't pose any real harm, but that doesn't undermine my point.

Reply:
Right won't do anything by default (i.e. with --video-recenter) in this case. There would no advantage to make it keep seeking, because seeking images doesn't do anything.

There are no plans to add other conditional profiles, images are a special case where it is worth it.

The options don't have side effects for videos, and previous values are restored when switching to a video.

Reply:
> What about holding RIGHT to skim over a video and when EOF has been reached and the player jumps to the next playlist item which is an image, the key suddenly changes its behavior?

What about holding `RIGHT` to skin over a video and when EOF has been reached and the player jumps to the next playlist item and they accidentally skim through the next file too? This can be a problem even on master itself, what you're asking for an option for mpv to release key presses when switching playlist item. Open an issue tracker so someone can work on it later. I don't see how it's related to this PR.

> My point is implementing conditional options intended for one type of files in a way that may have side effects for other types is flawed.

What are these side effects and what is flawed? Please elaborate. Do you mean to say `track-list/N/image` may be unreliable to detect images?

Reply:
> What are these side effects and what is flawed? Please elaborate.

Profile conditions are evaluated asynchronously, so there is a chance that options meant for images will affect other types of files shortly. And yes, I see that this is not likely to happen, but again, the possibility is there. If maintainers decide this can be tolerated, that's perfectly fine.

Reply:
> auto_profiles.lua already evaluates profiles within synchronous hooks, so applying the profile from C wouldn't magically make it more synchronous as far as I know, it just does the same thing with more code and more risk of memory issues.

OK somehow I missed your reply. If that's true then all my concerns have been addressed :)

Reply:
off-topic: why are we discussing this in a sub-thread attached to a single line change? it's inconvenient

<hr>

> libplacebo supports sliced texel uploads on d3d11 and vulkan, and mpv no longer defaults to opengl on any platform so this shouldn't be an issue.

I don't see how that's related since texture limits will still apply, won't they?

> neither does any image viewer using imlib2

okay? that just means those other image viewers are also bad

> Are you OK with enabling it globally by default?

If it works and has no other undesirable effects: sure why not.

> I never saw even one of these images bigger than GL max texture sizes in years of real world usage,

I agree it's not very common but I have wanted to view big images in real usage.

> image-input.conf is a workaround to not use deprecated input sections, which let you specify key bindings with {image} in the same input.conf

AFAIK input sections are only deprecated for what users do with mpv. there's no reason we can't add new features inside mpv that rely on them, for example encoding does too.

> Or what usefulness does seeking with arrow keys have on an image file with 1 frame?

see sunpenghao's response for this.
Introducing a dynamically switched second set of controls has to be done carefully to avoid confusing situations. Not saying we can't do it.

> There would no advantage to make it keep seeking, because seeking images doesn't do anything.

Doesn't it go to the next playlist item when you reach EOF?

> What about holding RIGHT to skin over a video and when EOF has been reached and the player jumps to the next playlist item and they accidentally skim through the next file too?

What? This is literally working as designed, because the RIGHT key has a defined meaning and it keeps getting applied if you hold the key.

<hr>

anyway:
Personally I think a dedicated image viewing mode is scope creep and I would rather suggest making it an user script and maybe prominently advertising it, but *not* including it in mpv core.